Starcounter.Uniform.Tests/Builder/UniDataTableTests.cs

```diff
@@ -0,0 +1,155 @@
+        }
+
+        [Test]
+        public void AfterInitSpecifiedColumnShouldBePresent()
+        {
+            var columnDisplayName = "First name";
+            _dataTableColumns.Add(new DataTableColumn { DisplayName = columnDisplayName });
+
+            InitSut();
+
+            _sut.Columns.Should().HaveCount(1);
+            _sut.Columns.Should().ContainSingle().Which.DisplayName.Should().Be(columnDisplayName);
+            _sut.Columns.Should().ContainSingle().Which.LoadRows.Should().NotBeNull();
+        }
+
+        [Test]
+        public void AfterInitInitialPageShouldBeLoaded()
+        {
+            var expectedRow = new Json();
+            var expectedRowsCollection = new List<Json>
+            {
+                expectedRow
+            };
+            var initialPageIndex = 3;
+            _initialPageIndex = initialPageIndex;
+            _dataProviderMock
+                .Setup(provider => provider.CurrentPageRows)
+                .Returns(() =>
+                _dataProviderMock.Object.PaginationConfiguration
+                    .CurrentPageIndex == initialPageIndex
+                    ? expectedRowsCollection
+                    : new List<Json>());
+
+            InitSut();
+
+            _sut.Pages.Should().HaveCount(initialPageIndex + 1);
+            _sut.Pages[initialPageIndex].Rows.Should().ContainSingle().Which.Should().BeSameAs(expectedRow);
+        }
+
+        [Test]
+        public void AfterChangingCurrentPageIndexPaginationConfigurationShouldChange()
+        {
+            var newPageIndex = 2;
+
+            InitSut();
+            _sut.Pagination.Handle(new UniDataTable.PaginationViewModel.Input.CurrentPageIndex { Value = newPageIndex });
+
+            _dataProviderMock.Object.PaginationConfiguration.CurrentPageIndex.Should().Be(newPageIndex);
+            _dataProviderMock.VerifyGet(x => x.CurrentPageRows, Times.Exactly(2));
+        }
+
+        [Test]
+        public void AfterCallingFilterHandleFilterValueInFilterOrderConfigurationShouldChange()
+        {
+            var expectedFilterValue = "John";
+            _dataTableColumns.Add(new DataTableColumn { PropertyName = _columnPropertyName });
+            var filter = new Filter { PropertyName = _columnPropertyName, Value = "Ann" };
+
+            InitSut();
+            _dataProviderMock.Object.FilterOrderConfiguration.Filters.Add(filter);
+            var firstNameColumn = _sut.Columns.First(x => x.PropertyName == _columnPropertyName);
+            firstNameColumn.Handle(new UniDataTable.ColumnsViewModel.Input.Filter { Value = expectedFilterValue });
+
+            _dataProviderMock.Object.FilterOrderConfiguration.Filters.Should().ContainSingle(x => x.PropertyName == _columnPropertyName).Which.Value.Should()
+                .Be(expectedFilterValue);
+        }
+
+        [Test]
+        public void AfterCallingFilterHandleForNotExistingFilterNewFilterShouldBeAdded()
```

Comment:
I think what you're testing here is that existing filter would be overridden. But if that is so, then:

1. make it more clear in your test name and variables (`oldFilter`?)
1. test explicitly that previous filter was deleted from `FilterOrderConfiguration`
1. test that previous filter was deleted from the view-model

Reply:
No, here I'm testing exactly what test name says, I'm checking if after calling filter handle for filter that do not exists, new filter is created in filters collection.